pkg/state/state_doca_telemetry_service.go

```diff
@@ -0,0 +1,196 @@
+/*
+2024 NVIDIA CORPORATION & AFFILIATES
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	mellanoxv1alpha1 "github.com/Mellanox/network-operator/api/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/config"
+	"github.com/Mellanox/network-operator/pkg/consts"
+	"github.com/Mellanox/network-operator/pkg/render"
+	"github.com/Mellanox/network-operator/pkg/utils"
+)
+
+type docaTelemetryServiceState struct {
+	stateSkel
+}
+
+type dtsRuntimeSpec struct {
+	runtimeSpec
+	ContainerResources ContainerResourcesMap
+}
+
+const (
+	docaTelemetryServiceName                 = "state-doca-telemetry-service"
+	docaTelemetryServiceDefaultConfigMapName = "doca-telemetry-service"
+	docaTelemetryServiceDescription          = "DOCA Telemetry Service deployed in the cluster"
+)
+
+// DOCATelemetryServiceManifestRenderData is used to render Kubernetes objects related to DOCA Telemetry Service.
+type DOCATelemetryServiceManifestRenderData struct {
+	CrSpec        *mellanoxv1alpha1.DOCATelemetryServiceSpec
+	ConfigMapName string
+	RuntimeSpec   *dtsRuntimeSpec
+	Tolerations   []v1.Toleration
+	NodeAffinity  *v1.NodeAffinity
+}
+
+// Sync attempt to get the system to match the desired state which State represents.
+// a sync operation must be relatively short and must not block the execution thread.
+func (d docaTelemetryServiceState) Sync(
```

Comment:
any plans to add unit test for it ? like we started to do with different states ?

Reply:
Yeah - it's on the backlog. Right now I'd like to get #814 merged before moving ahead with unit tests here as that could simplify things alot.

Reply:
Happy to do it as part of this PR - but would like to know if I can do them in the style from 814 first.